### PR TITLE
Add --create-application-cr flag to create Application CR and --links flag to set Application links.

### PR DESCRIPTION
**Pull request by joonlim**

### CODE REVIEW

**Comment by joonlim** [comment]
@gleeper could you please look at the commits that are not "Refresh vendor, go.mod, go.sum"

**Comment by joonlim** [comment]
@chrisge4

File: gke-deploy/README.md [hunk 1]
```diff
@@ -52,6 +52,21 @@ View [this page](doc/deploying-with-cloud-build.md) for examples on how to use
 Follow [these instructions](doc/automated-deployments.md) to set up continuous
 deployment.
 
+## Application CR
+
+gke-deploy supports deploying your application with an Application
```
Comment (dorbin) [inline gke-deploy/README.md]:
```suggestion
You can also deploy your application using an Application
```

File: gke-deploy/README.md [hunk 2]
```diff
@@ -52,6 +52,21 @@ View [this page](doc/deploying-with-cloud-build.md) for examples on how to use
...
+[Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
```
Comment (dorbin) [inline gke-deploy/README.md]:
```suggestion
[Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/):
```

File: gke-deploy/README.md [hunk 3]
```diff
@@ -52,6 +52,21 @@ View [this page](doc/deploying-with-cloud-build.md) for examples on how to use
...
+This requires your GKE cluster to have the Application CRD installed. Follow the
```
Comment (dorbin) [inline gke-deploy/README.md]:
```suggestion

1. Follow the
```

File: gke-deploy/README.md [hunk 4]
```diff
@@ -52,6 +52,21 @@ View [this page](doc/deploying-with-cloud-build.md) for examples on how to use
...
+instructions
+[here](https://cloud-dot-devsite.googleplex.com/kubernetes-engine/docs/how-to/add-on/application-delivery)
+to create or update a GKE cluster to support the Application
+Delivery, which supports the Application CRD, or install the CRD manually by
```
Comment (dorbin) [inline gke-deploy/README.md]:
```suggestion
Delivery, which supports the Application CRD.

  You can also install the CRD manually by following the instructions [here](https://github.com/kubernetes-sigs/application#installing-the-crd).
```

File: gke-deploy/README.md [hunk 5]
```diff
@@ -52,6 +52,21 @@ View [this page](doc/deploying-with-cloud-build.md) for examples on how to use
...
+following the instructions
```
Comment (dorbin) [inline gke-deploy/README.md]:
```suggestion
```

File: gke-deploy/README.md [hunk 6]
```diff
@@ -52,6 +52,21 @@ View [this page](doc/deploying-with-cloud-build.md) for examples on how to use
...
+[here](https://github.com/kubernetes-sigs/application#installing-the-crd).
```
Comment (dorbin) [inline gke-deploy/README.md]:
```suggestion
```

File: gke-deploy/README.md [hunk 7]
```diff
@@ -52,6 +52,21 @@ View [this page](doc/deploying-with-cloud-build.md) for examples on how to use
...
+
+Use the `--create-application-cr` flag with `gke-deploy prepare` or `gke-deploy
```
Comment (dorbin) [inline gke-deploy/README.md]:
```suggestion
1. Use the `--create-application-cr` flag with `gke-deploy prepare` or `gke-deploy
```

File: gke-deploy/doc/gke-deploy_prepare.md [hunk 8]
```diff
@@ -35,22 +35,24 @@ gke-deploy prepare [flags]
...
-  -o, --output string        Target directory to store suggested and expanded Kubernetes configuration files. Suggested files will be stored in "<output>/suggested" and expanded files will be stored in "<output>/expanded". (default "./output")
-  -R, --recursive            Recursively search through the configuration directory for all yaml files.
-  -V, --verbose              Prints underlying commands being called to stdout.
-  -v, --version string       Version of the Kubernetes deployment.
+  -A, --annotation strings      Annotation(s) to add to Kubernetes configuration files (k1=v1). Annotations can be set comma-delimited or as separate flags. If two or more annotations with the same key are listed, the last one is used.
+  -a, --app string              Application name of the Kubernetes deployment.
+      --create-application-cr   Creates an Application CR object with the name provided by --app and connects to a deployed objects using a selector that matches the label with key as 'app.kubernetes.io/name' and value specified by --app.
```
Comment (dorbin) [inline gke-deploy/doc/gke-deploy_prepare.md]: Either "objects" should be "object" here (singular), or delete the "a"

File: gke-deploy/doc/gke-deploy_run.md [hunk 9]
```diff
@@ -41,26 +41,28 @@ gke-deploy run [flags]
...
-  -p, --project string       Project of GKE cluster to deploy to. If this field is not provided, the current set GCP project is used.
-  -R, --recursive            Recursively search through the configuration directory for all yaml files.
-  -t, --timeout duration     Timeout limit for waiting for Kubernetes objects to finish applying. (default 5m0s)
-  -V, --verbose              Prints underlying commands being called to stdout.
-  -v, --version string       Version of the Kubernetes deployment.
+  -A, --annotation strings      Annotation(s) to add to Kubernetes configuration files (k1=v1). Annotations can be set comma-delimited or as separate flags. If two or more annotations with the same key are listed, the last one is used.
+  -a, --app string              Application name of the Kubernetes deployment.
+  -c, --cluster string          Name of GKE cluster to deploy to.
+      --create-application-cr   Creates an Application CR object with the name provided by --app and connects to a deployed objects using a selector that matches the label with key as 'app.kubernetes.io/name' and value specified by --app.
```
Comment (dorbin) [inline gke-deploy/doc/gke-deploy_run.md]: Same thing here; shouldn't be "...a deployed objects"

File: gke-deploy/services/remote.go [hunk 10]
```diff
@@ -17,7 +17,7 @@ import (
 
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
-	"github.com/google/go-containerregistry/pkg/v1"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
```
Comment (ChrisGe4) [inline gke-deploy/services/remote.go]: Leave a comment to remind you to revert this change.
